team-builder/src/App.js

```diff
@@ -0,0 +1,150 @@
+import React, {useState} from 'react';
+import uuid from 'uuid';
+import './App.css';
+
+const initialTeamList = [
+  // why is it useful to have a unique id?
+  { id: uuid(), name: 'olu', email: 'olu@email.com', role: 'UI Dev'},
+  { id: uuid(), name: 'tomiwa', email: 'tomiwa@email.com', role: 'Javascript Dev'},
+  { id: uuid(), name: 'andrew', email: 'andrew@email.com', role: 'Python Dev'},
+];
+
+const initialTeamForm = {
+  name: '',
+  email: '',
+  role:'',
+};
+
+
+export default function App() {
+
+  const [teamList,setTeamList]= useState(initialTeamList);
+  const [teamForm,setTeamForm]=useState(initialTeamForm);
+
+  const onNameChange = event => {
+    // setTeamForm({...teamList, name: event.target.value})
+    setTeamForm({
+      name: event.target.value,
+      email:teamForm.email,
+      role:teamForm.role,
+    })
+  };
+
+  const onEmailChange = event => {
+    // setTeamForm({...teamList, email: event.target.value})
+    setTeamForm({
+      name: teamForm.name,
+      email: event.target.value,
+      role:teamForm.role,
+    })
+  };
+
+  const onRoleChange = event => {
+    // setTeamForm({...teamList, role: event.target.value})
+    setTeamForm({
+      name: teamForm.name,
+      email: teamForm.role,
```

Comment:
`email: teamForm.role` gets overwritten by the changes in `role` input field. Just a typo, but be careful 👍

Reply:
Oh! Thanks.